Files already downloaded and verified
Files already downloaded and verified
Data:
 - Train Set: ((50000, 3, 32, 32),(50000,)), Range: [0.000, 255.000], Labels: 0,..,9
 - Test Set: ((10000, 3, 32, 32),(10000,)), Range: [0.000, 255.000], Labels: 0,..,9
Data split:
 - Client 0: [252 237 251 257 260 238 239 269 270 227]
 - Client 1: [257 251 257 236 244 258 249 225 270 253]
 - Client 2: [246 213 237 273 271 251 264 255 248 242]
 - Client 3: [244 233 256 249 245 247 295 244 251 236]
 - Client 4: [269 239 232 252 260 223 271 250 257 247]
 - Client 5: [253 292 268 238 254 238 249 225 232 251]
 - Client 6: [250 268 262 258 234 248 230 233 251 266]
 - Client 7: [261 266 222 250 256 235 235 262 258 255]
 - Client 8: [256 233 276 271 243 242 245 252 240 242]
 - Client 9: [247 247 245 231 254 249 254 247 266 260]
 - Client 10: [   0    0  499    0    0    0    0    0 2001    0]
 - Client 11: [2465    0    0    0    0    0    0    0    0   35]
 - Client 12: [   0    0    0    0 2096    0    0    0  404    0]
 - Client 13: [   7    0    0    0 2493    0    0    0    0    0]
 - Client 14: [1610    0    0    0    0    0  890    0    0    0]
 - Client 15: [   0  165    0    0    0    0    0    0 2335    0]
 - Client 16: [ 788 1712    0    0    0    0    0    0    0    0]
 - Client 17: [   0    0    0    0    0    0 2447    0   53    0]
 - Client 18: [   0    0    0    0    0    0  383 2117    0    0]
 - Client 19: [   0    0    0  607    0    0    0 1893    0    0]
Traceback (most recent call last):
  File "train.py", line 57, in <module>
    train_loader, test_loader = get_data_loaders(classes_pc=classes_pc, nclients= num_clients,
  File "/home/zqf/projects/FLNONIID/ultils.py", line 317, in get_data_loaders
    split = split_balanced_non_iid(x_train, y_train, n_clients=nclients, n_iid=10)
  File "/home/zqf/projects/FLNONIID/ultils.py", line 388, in split_balanced_non_iid
    clients_split = np.array(clients_split, dtype=object)
ValueError: could not broadcast input array from shape (2500,3,32,32) into shape (2500,)